src/rulesets/merger.ts

```diff
@@ -86,3 +99,33 @@ function processRule(rules: FileRuleCollection, name: string, rule: FileRule) {
 function isValidRule(rule: FileRule): rule is Rule {
   return typeof rule === 'object' && rule !== null && !Array.isArray(rule) && ('given' in rule || 'then' in rule);
 }
+
+function getSeverityLevel(
+  rules: FileRuleCollection,
+  name: string,
+  rule: FileRule | FileRulesetSeverity,
+): DiagnosticSeverity | HumanReadableDiagnosticSeverity {
+  const existingRule = rules[name];
+
+  if (!isValidRule(existingRule)) return 'off';
+
+  if (rule === 'recommended') {
+    return getRootRule(existingRule).recommended
+      ? getRootRule(existingRule).severity || DiagnosticSeverity.Error // not sure what a default value could be
```

Comment:
@marbemac @philsturgeon 
I suppose we might need to mark `severity` as required.
Right now, both `all` and `recommended` don't make much sense, as severity might be missing, so we need to resort to treating them as errors.

Reply:
Sorry could you elaborate on this a bit?

Reply:
Sure, so what I'm trying to understand is which level of severity I should be set for inherited rules in the following case:

```
"extends": [[someRuleset, "recommended"]]
```
Severity does not have to be specified explicitly, so I have no clue whether an error level is a safe bet, or perhaps warning is a more suitable default.

Reply:
The `"extends": [[someRuleset, "recommended"]]` stuff is unrelated to severity - it should all be driven by the single `recommended` property.

Good question on severity though - this must have been an issue in v3? What did v3 do if severity was not set?

Reply:
> The "extends": [[someRuleset, "recommended"]] stuff is unrelated to severity - it should all be driven by the single recommended property.

Yeah, I get it - when recommended is set, I just inherit that rule. My main concern is that I don't set any severity and this is kind of weird. It definitely was an issue in v3, but right now, once we introduce human-readable severity levels, I try to normalize severity to match DiagnosticSeverity everywhere and I'm just stuck on that. I can leave it empty or so, but I find this to be kind of an issue in particular when you set the severity to `true` as it means you might end up trying to inherit a non-existing value t.

Reply:
Good point. Perhaps we should just have a default severity? Would solve it and is nice and easy. `warn`?

Reply:
Implemented in https://github.com/stoplightio/spectral/pull/232/commits/941c9cd2208fb5926eca1fae31ea011d110f4e90